satpy/readers/hy2_scat_l2b_h5.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""HY-2B L2B Reader, distributed by Eumetsat in HDF5 format
+"""
+
+import numpy as np
+import xarray as xr
+import dask.array as da
+from satpy import CHUNK_SIZE
+from datetime import datetime
+
+from satpy.readers.hdf5_utils import HDF5FileHandler
+
+
+class HY2SCATL2BH5FileHandler(HDF5FileHandler):
+
+    @property
+    def start_time(self):
+        """Time for first observation."""
+        return datetime.strptime(self['/attr/Range_Beginning_Time'],
+                                 '%Y%m%dT%H:%M:%S')
+
+    @property
+    def end_time(self):
+        """Time for final observation."""
+        return datetime.strptime(self['/attr/Range_Ending_Time'],
+                                 '%Y%m%dT%H:%M:%S')
+
+    def get_metadata(self):
```

Comment:
Hm, some of the attributes are most likely the same, but others will change from file to file. Then it will be necessary to group this into two function, one with cache_property and the other not.

Reply:
Yes, but the cached property should be per instance (so per file)

Reply:
Ah ok. I might need some help to get me in the right direction.

Reply:
ok, turns out `cached_property` is only available from python 3.8, so we can't use it right now.